setup.py

```diff
@@ -0,0 +1,146 @@
+    ]
+
+    def initialize_options(self):
+        self.target = "sonata_python"
+        super(CMakeBuild, self).initialize_options()
+
+    def run(self):
+        try:
+            out = subprocess.check_output(["cmake", "--version"])
+        except OSError:
+            raise RuntimeError(
+                "CMake must be installed to build the following extensions: "
+                + ", ".join(e.name for e in self.extensions)
+            )
+
+        if platform.system() == "Windows":
+            cmake_version = LooseVersion(
+                re.search(r"version\s*([\d.]+)", out.decode()).group(1)
+            )
+            if cmake_version < "3.1.0":
+                raise RuntimeError("CMake >= 3.1.0 is required on Windows")
+
+        for ext in self.extensions:
+            self.build_extension(ext)
+
+    def build_extension(self, ext):
+        extdir = osp.abspath(osp.dirname(self.get_ext_fullpath(ext.name)))
+        # extdir = osp.join(extdir, 'libsonata')
+        print("extdir: {}".format(extdir))
+        print("CMAKE_LIBRARY_OUTPUT_DIRECTORY=" + extdir)
+        cmake_args = [
+            "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=" + extdir,
+            "-DEXTLIB_FROM_SUBMODULES=ON",
+            "-DSONATA_PYTHON=ON",
+            "-DCMAKE_BUILD_TYPE=",
+        ]
+
+        optimize = "OFF" if self.debug else "ON"
+        build_args = ["--config", optimize, "--target", self.target]
+
+        if platform.system() == "Windows":
+            cmake_args += [
+                "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(
+                    optimize.upper(), extdir
+                )
+            ]
+            if sys.maxsize > 2 ** 32:
+                cmake_args += ["-A", "x64"]
+            build_args += ["--", "/m"]
+        else:
+            build_args += ["--", "-j{}".format(max(1, os.cpu_count() - 1))]
+
+        env = os.environ.copy()
+        env["CXXFLAGS"] = '{} -DVERSION_INFO=\\"{}\\"'.format(
+            env.get("CXXFLAGS", ""), self.distribution.get_version()
+        )
+        if not osp.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+        subprocess.check_call(
+            ["cmake", ext.sourcedir] + cmake_args, cwd=self.build_temp, env=env
+        )
+        subprocess.check_call(
+            ["cmake", "--build", "."] + build_args, cwd=self.build_temp
+        )
+
+
+class PkgTest(test):
+    """Custom disutils command that acts like as a replacement
+    for the "test" command.
+    """
+
+    new_commands = [('test_ext', lambda self: True), ('test_doc', lambda self: True)]
+    sub_commands = test.sub_commands + new_commands
+
+    def run(self):
+        super().run()
+        self.run_command('test_ext')
+        self.run_command('test_doc')
+
+install_requirements = [
+    "numpy>=1.12.0",
+]
+
+setup(
+    name="libsonata",
+    description='SONATA files reader',
+    author="BlueBrain Project, EPFL",
+    author_email="bbp-ou-hpc@groupes.epfl.ch",
```

Comment:
Let's keep NSE mailing list for the moment.

For the rest of the file, I assume it is taken from something already working (and reviewed), so just take it for granted. :)